src/read.js

```diff
@@ -150,8 +171,8 @@ class Entry {
   }
 
   get size() {
-    const size = this.dataView.getUint32(24, true);
-    return size === MAX_VALUE_32BITS ? this._extraFields[1].getUint8(0) : size;
+    const size = this.uncompressedSize;
+    return size === MAX_VALUE_32BITS ? Number(getBigInt64(this._extraFields[0x0001], 0)) : size;
```

Comment:
To keep consistency with other calls.

```suggestion
    return size === MAX_VALUE_32BITS ? JSBI.toNumber(getBigInt64(this._extraFields[0x0001], 0)) : size;
```